src/ApplicationManager.cpp

```diff
@@ -35,32 +35,44 @@ void ApplicationManager::loop() {
     int nTicksTaken = 0;
     double timeTakenOverNTicks = 0.0;
 
+    std::chrono::milliseconds now_time;
+    std::chrono::milliseconds last_call_time = now_time;
+    std::chrono::milliseconds last_fps_count_time = now_time;
 
-    std::chrono::milliseconds ms = std::chrono::duration_cast< std::chrono::milliseconds >(
-        std::chrono::system_clock::now().time_since_epoch()
-    );
+    bool ok = true;
 
-    std::chrono::milliseconds last_call_time = ms;
+    int cyclesInThisSecond = 0;
 
-    bool ok = true;
+    int lastFPS = 0;
     while(ok) {
 
-        std::chrono::milliseconds ms = std::chrono::duration_cast< std::chrono::milliseconds >(
+        now_time = std::chrono::duration_cast< std::chrono::milliseconds >(
             std::chrono::system_clock::now().time_since_epoch()
         );
-        std::chrono::milliseconds now_time = ms;
-
         auto diff = now_time - last_call_time;
-
         auto timeDiff =std::chrono::milliseconds(16); //a little over 60hz
-
         if(diff > timeDiff) {
-
             this->runOneLoopCycle();
+
+            std::chrono::milliseconds afterOneCycleTime = std::chrono::duration_cast< std::chrono::milliseconds >(
+                    std::chrono::system_clock::now().time_since_epoch());
+
+            cyclesInThisSecond++;
+
+            auto fps_diff = now_time - last_fps_count_time;
+            auto fps_timestep =std::chrono::milliseconds(200); // one second
```

Comment:
I would maybe make sure that this value (200 ms) is documented clearly somewhere, since we had issues with not detecting problems with 1 second intervals. I wouldn't mind this dropping to 100 ms or even 50 ms so problems are easier to see.